src/Compilers/CSharp/Test/Emit3/OverloadResolutionPriorityTests.cs

```diff
@@ -0,0 +1,1591 @@
+
+                [OverloadResolutionPriority(0)]
+                public static void M(I2 x) => throw null;
+            }
+            """;
+        CreateCompilation([executable, source, OverloadResolutionPriorityAttributeDefinition]).VerifyDiagnostics(
+            // (2,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(I1)' and 'C.M(I2)'
+            // C.M(i3);
+            Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(I1)", "C.M(I2)").WithLocation(2, 3)
+        );
+
+        var comp = CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]);
+        CompileAndVerify(comp, symbolValidator: module =>
+        {
+            var c = module.GlobalNamespace.GetMember<NamedTypeSymbol>("C");
+            var ms = c.GetMembers("M").Cast<MethodSymbol>();
+            Assert.All(ms, m => Assert.Equal(0, m.OverloadResolutionPriority));
+        }).VerifyDiagnostics();
+
+        CreateCompilation(executable, references: [useMetadataReference ? comp.ToMetadataReference() : comp.EmitToImageReference()]).VerifyDiagnostics(
+            // (2,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(I1)' and 'C.M(I2)'
+            // C.M(i3);
+            Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(I1)", "C.M(I2)").WithLocation(2, 3)
+        );
+    }
+
+    [Theory]
+    [InlineData(-1)]
+    [InlineData(1)]
+    [InlineData(int.MaxValue)]
+    [InlineData(int.MinValue)]
+    public void AmbiguityWithinPriority(int priority)
+    {
+        var source = $$"""
+            using System.Runtime.CompilerServices;
+
+            I3 i3 = null;
+            C.M(i3);
+
+            interface I1 {}
+            interface I2 {}
+            interface I3 : I1, I2 {}
+
+            class C
+            {
+                [OverloadResolutionPriority({{priority}})]
+                public static void M(I1 x) => System.Console.WriteLine(1);
+
+                [OverloadResolutionPriority({{priority}})]
+                public static void M(I2 x) => throw null;
+            }
+            """;
+
+        CreateCompilation([source, OverloadResolutionPriorityAttributeDefinition]).VerifyDiagnostics(
+            // (4,3): error CS0121: The call is ambiguous between the following methods or properties: 'C.M(I1)' and 'C.M(I2)'
+            // C.M(i3);
+            Diagnostic(ErrorCode.ERR_AmbigCall, "M").WithArguments("C.M(I1)", "C.M(I2)").WithLocation(4, 3)
+        );
+    }
+
+    [Theory, CombinatorialData]
+    public void MethodDiscoveryStopsAtFirstApplicableMethod(bool useMetadataReference)
+    {
+        var @base = """
+            using System.Runtime.CompilerServices;
+
+            public interface I1 {}
+            public interface I2 {}
+            public interface I3 : I1, I2 {}
+
+            public class Base
+            {
+                [OverloadResolutionPriority(1)]
+                public static void M(I2 x) => throw null;
```

Comment:
>public static void M(I2 x) => throw null; [](http://example.com/codeflow?start=16&length=41)

Isn't this method dropped from the candidate set even before we get to overload resolution priority?

Reply:
Yes, that's the point of this test.